odoo/addons/base/models/res_users.py

```diff
@@ -394,9 +394,50 @@ def _check_action_id(self):
     @api.multi
     @api.constrains('groups_id')
     def _check_one_user_type(self):
-        for user in self:
-            if len(user.groups_id.filtered(lambda x: x.category_id.xml_id == 'base.module_category_user_type')) > 1:
-                raise ValidationError(_('The user cannot have more than one user types.'))
+        """We check that no users are both portal and users (same with public).
+           This could typically happen because of implied groups.
+        """
+        user_types_category = self.env.ref('base.module_category_user_type', raise_if_not_found=False)
+        user_types_groups = self.env['res.groups'].search(
+            [('category_id', '=', user_types_category.id)]) if user_types_category else False
+        if user_types_groups:  # needed at install
+            bad_users = self._fast_check_users_in_more_than_one_group(user_types_groups.ids)
+            if bad_users:
+                raise ValidationError(_('The user cannot have more than one user types.') + '\n' +
+                                      _('Problematic ids: ') + ", ".join(str(bad[0]) for bad in bad_users))
```

Comment:
Limit the size of the list to avoid huge strings if thousands of users are problematic.

Reply:
Ditto for showing the IDs